Maskbook/Scene/App/FIleService/Model/FileServiceSelectFileHandler.swift

```diff
@@ -0,0 +1,188 @@
+        self.delegate = delegate
+    }
+
+    @InjectedProvider(\.mainCoordinator)
+    private var coordinator
+
+    var uploadFolder: URL {
+        let path = URL.documents.appendingPathComponent("fileServiceUpload")
+        var isDirectory: ObjCBool = true
+        if !FileManager.default.fileExists(atPath: path.path, isDirectory: &isDirectory) {
+            try? FileManager.default.createDirectory(at: path, withIntermediateDirectories: true)
+        }
+        return path
+    }
+
+    var delegate: FileServiceSelectFileDelegate
+    private(set) lazy var imagePicker: PHPickerViewController = {
+        var configuration = PHPickerConfiguration()
+        configuration.filter = .images
+        configuration.selectionLimit = 1
+
+        let imagePicker = PHPickerViewController(configuration: configuration)
+        imagePicker.delegate = self
+        return imagePicker
+    }()
+
+    private(set) lazy var cameraController: UIImagePickerController = {
+        let imagePickerController = UIImagePickerController()
+        imagePickerController.sourceType = .camera
+        imagePickerController.delegate = self
+        return imagePickerController
+    }()
+
+    private(set) lazy var documentPicker: UIDocumentPickerViewController = {
+        if #available(iOS 15.0, *) {
+            return UIDocumentPickerViewController(forOpeningContentTypes: [.data])
+        } else {
+            return UIDocumentPickerViewController(documentTypes: ["public.data"], in: .open)
+        }
+    }()
+
+    func select(item: FileServiceSelectFileSourceViewModel.LocalFileSourceItem) {
+        coordinator.dismissTopViewController(animated: false) {
+            switch item {
+            case .camera:
+                self.takePhotoAction()
+            case .photo:
+                self.selectFromPhotosAction()
+            case .file:
+                self.selectFormFilesAction()
+            }
+        }
+    }
+
+    func randomNameForImage() -> String {
+        let randomString = UUID().uuidString
+        return "userImage" + String(randomString.prefix(5)) + ".jpg"
+    }
+
+    func saveImageToLocal(image: UIImage) {
+        if let pngData = image.jpeg(UIImage.JPEGQuality.high), !pngData.isEmpty {
+            let fileName = randomNameForImage()
+            let path = uploadFolder.appendingPathComponent(fileName)
+            try? pngData.write(to: path)
+            delegate.didGetImage(image: image, fileURL: path, fileName: fileName)
+        } else {
+            assertionFailure("can't get jpeg data from image")
+        }
+    }
+
+    func didGetImage(image: UIImage) {
+        saveImageToLocal(image: image)
+    }
+
+    func didGetFile(url: URL) {
+        delegate.didGetFile(url: url)
+    }
+}
+
+extension FileServiceSelectFileHandler: PHPickerViewControllerDelegate {
+    func picker(_ picker: PHPickerViewController, didFinishPicking results: [PHPickerResult]) {
+        DispatchQueue.main.async {
```

Comment:
Can we refactor this?
1. Avoid  nested `DispatchQueue.main.async` call.
2. Abstract getting `image` for `results` into a function and return a `Result<UIImage, Error>`.

Reply:
1、how to Avoid nested DispatchQueue.main.async call in this method?